### PR TITLE
Avoid unnecessary user get expiring session memberships

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This is something to speed up our middleware by 1 database query for every request.

##### ISSUE TYPE
 - Performance

##### COMPONENT NAME
 - API

##### AWX VERSION
```
4.0.0
```

##### ADDITIONAL INFORMATION
there seems to be no need to obtain the object at all, so we can just not do it. I tested the filter and create commands via the shell manually, and I know we have at least 1 functional unit test that covers this code path.

### CODE REVIEW

**Comment by AlanCoding** [comment]
recheck

**Review by rooftopcellist — APPROVED** [review]
lgtm

**Comment by AlanCoding** [comment]
I have a gut feeling that I should rebase and test this with https://github.com/ansible/awx/pull/3875 before merging it, so I will call it blocked until that lands. This makes the most sense in terms of simple merge order.

**Comment by AlanCoding** [comment]
recheck

**Comment by AlanCoding** [comment]
Also want to test with rebase on top of https://github.com/ansible/awx/pull/3918 and https://github.com/ansible/awx/pull/3890

**Comment by AlanCoding** [comment]
This is kind of the gist of the answer I was looking for:

```
/venv/awx/lib64/python3.6/site-packages/django/contrib/sessions/middleware.py in process_response(58)
  request.session.save()
/venv/awx/lib64/python3.6/site-packages/django/contrib/sessions/backends/db.py in save(87)
  obj.save(force_insert=must_create, force_update=not must_create, using=using)
/venv/awx/lib64/python3.6/site-packages/django/dispatch/dispatcher.py in send(193)
  for receiver in self._live_receivers(sender)
/venv/awx/lib64/python3.6/site-packages/django/dispatch/dispatcher.py in <listcomp>(193)
  for receiver in self._live_receivers(sender)
./awx/main/signals.py in save_user_session_membership(658)
  if UserSessionMembership.objects.filter(user=user_id, session=session).exists():
```

I just wanted to have a sense of _if_ and _where_ this gets called from after the reordering. It seems that it is called from `SessionMiddleware`

https://github.com/django/django/blob/1.11.19/django/contrib/sessions/middleware.py#L58

Everything should be perfectly okay with that.

This is a post-save signal, and we are able to clearly identify the original save that happens

```
/venv/awx/lib64/python3.6/site-packages/django/contrib/sessions/middleware.py in process_response(58)
  request.session.save()
/venv/awx/lib64/python3.6/site-packages/django/contrib/sessions/backends/db.py in save(87)
  obj.save(force_insert=must_create, force_update=not must_create, using=using)
```

**Pull request merged**
